src/sc/sidechain.cpp

```diff
@@ -757,61 +778,139 @@ void CSidechain::UpdateScFees(const CScCertificateView& certView)
     LogPrint("sc", "%s():%d - pushing f=%d/m=%d into list with size %d\n",
         __func__, __LINE__, ftScFee, mbtrScFee, scFees.size());
 
-    scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
+    // v1 sidechains
+    if (!isNonCeasing()) {
+        scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
 
-    // check if we are past the buffer size
-    int delta = scFees.size() - maxSizeOfScFeesContainers;
-    if (delta > 0)
-    {
-        // remove from the front as many elements are needed to be within the circular buffer size
-        // --
-        // usually this is just one element, but in regtest a node can set the max size via a startup option
-        // therefore such size might be lesser than scFee size after a node restart
-        while (delta > 0)
+        int delta = scFees.size() - maxSizeOfScFeesContainers;
+        if (delta > 0)
         {
-            if (scFees.empty())
-                break;
+            // remove from the front as many elements are needed to be within the circular buffer size
+            // --
+            // usually this is just one element, but in regtest a node can set the max size via a startup option
+            // therefore such size might be lesser than scFee size after a node restart
+            while (delta > 0)
+            {
+                if (scFees.empty())
+                    break;
+
+                const auto& entry = scFees.front();
+                LogPrint("sc", "%s():%d - popping f=%d, m=%d from list, as scFees maxsize has been reached\n",
+                    __func__, __LINE__, entry.forwardTxScFee, entry.mbtrTxScFee);
+                scFees.pop_front();
+                delta--;
```

Comment:
Prefer pre-increments rather than post-increments when value is not immediately used. 

Rationale https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Reply:
All occurrences of `delta--` replaced with `--delta` in f0bab99cd7559a9bab4c8dc65103df6eea2b4b76